File: server/src/main/java/org/dan/jadalnia/app/festival/pojo/Festival.java
package org.dan.jadalnia.app.festival.pojo;

import com.google.common.util.concurrent.AtomicLongMap;
import lombok.AccessLevel;
import lombok.Builder;
import lombok.Getter;
import lombok.Setter;
import lombok.experimental.FieldDefaults;
import org.dan.jadalnia.app.festival.order.pojo.PaidOrder;
import org.dan.jadalnia.app.festival.menu.DishName;
import org.dan.jadalnia.app.festival.order.pojo.Oid;
import org.dan.jadalnia.app.user.Uid;

import java.util.LinkedHashMap;
import java.util.Map;
import java.util.concurrent.atomic.AtomicReference;

@Getter
@Setter
@Builder
@FieldDefaults(level = AccessLevel.PRIVATE, makeFinal = true)
public class Festival {
    public static final String FESTIVAL_STATE = "state";
    public static final String TID = "fid";

    AtomicReference<FestivalInfo> info;

    LinkedHashMap<Oid, PaidOrder> paidOrders;
    Map<Oid, Uid> kelnersProcessingOrders;

    AtomicLongMap<DishName> requiredItems;
}

File: kot/src/main/kotlin/org/dan/jadalnia/app/festival/pojo/Festival.kt
package org.dan.jadalnia.app.festival.pojo;


import org.dan.jadalnia.app.order.line.OrderExecEstimationState
import org.dan.jadalnia.app.order.pojo.OrderLabel
import org.dan.jadalnia.app.user.Uid
import org.dan.jadalnia.util.collection.MapQ
import java.util.concurrent.ConcurrentMap
import java.util.concurrent.atomic.AtomicInteger
import java.util.concurrent.atomic.AtomicReference

class Festival(
    val info: AtomicReference<FestivalInfo>,
    val readyToExecOrders: MapQ<OrderLabel>,
    val readyToPickupOrders: ConcurrentMap<OrderLabel, Unit>,
    val executingOrders: ConcurrentMap<OrderLabel, Uid>,
    val freeKelners: ConcurrentMap<Uid, FreeKelnerInfo>,
    val busyKelners: ConcurrentMap<Uid, OrderLabel>,
    val nextLabel: AtomicInteger,
    val queuesForMissingMeals: MapOfQueues,
    val estimatorState: OrderExecEstimationState,
    val nextToken: AtomicInteger) {

    fun fid() = info.get().fid
}
